src/components/table/DatasetTable.jsx

```diff
@@ -21,21 +21,38 @@ class DatasetTable extends React.PureComponent {
     classes: PropTypes.object.isRequired,
     datasets: PropTypes.array.isRequired,
     datasetsCount: PropTypes.number,
+    features: PropTypes.object,
     handleFilterDatasets: PropTypes.func,
     summary: PropTypes.bool,
   };
 
   render() {
-    const { classes, datasets, datasetsCount, handleFilterDatasets, summary } = this.props;
+    const {
+      classes,
+      datasets,
+      datasetsCount,
+      features,
+      handleFilterDatasets,
+      summary,
+    } = this.props;
     // TODO add back modified_date column
     const columns = [
       {
         label: 'Dataset Name',
         property: 'name',
         render: (row) => (
-          <Link to={`/datasets/details/${row.id}`} className={classes.jadeLink}>
-            {row.name}
-          </Link>
+          <div>
+            {features.granular_sharing && (
+              <Link to={`/datasets/${row.id}/details`} className={classes.jadeLink}>
```

Comment:
You might be able condense this down to just a ternary condition in the `to` prop